### PR TITLE
Make it possible to configure the cacheMethod hashing algorithm in DboSource

### PR DESCRIPTION
**Pull request by tersmitten**
> Method caching uses `md5` to construct cache keys. If you have problems with collisions, set DboSource::$cacheMethods to false.

Seems like an easy fix for this problem. If you like this feature I can add some tests.

Thanks in advance

### CODE REVIEW

File: lib/Cake/Model/Datasource/DboSource.php [hunk 1]
```diff
@@ -51,23 +51,36 @@ class DboSource extends DataSource {
...
 
 /**
+ * Method caching uses `md5` (by default) to construct cache keys. If you have problems with collisions,
+ * try a different hashing algorithm or set DboSource::$cacheMethods to false.
+ *
+ * @var string
+ * @see http://php.net/manual/en/function.hash-algos.php
+ * @see http://softwareengineering.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed
+ */
+	public $cacheMethodHashAlgo = 'md5';
```
Comment (markstory) [inline lib/Cake/Model/Datasource/DboSource.php]: Perhaps another place to use a method instead of a property?

Comment (tersmitten) [inline lib/Cake/Model/Datasource/DboSource.php]: That would make it more flexible

**Comment by tersmitten** [comment]
I think this is ready for merging

**Comment by tersmitten** [comment]
I'll fix the conflicts

**Comment by tersmitten** [comment]
Done

**Pull request merged**